elements/power.lua

```diff
@@ -208,9 +212,23 @@ local Update = function(self, event, unit)
         local adjust = 0 - (min or 0)
 		r, g, b = self.ColorGradient(cur + adjust, max + adjust, unpack(power.smoothGradient or self.colors.smooth))
 	end
-
+	
+	if(power.statusbar) then
+		statusbar = power.statusbar
+	end
+	
 	if(t) then
-		r, g, b = t[1], t[2], t[3]
+		if ( power.colorPower and power.atlasEnable ) then
+			if ( t.atlas ) then
+				power:SetStatusBarAtlas(t.atlas);
+				power:SetStatusBarColor(1, 1, 1);
+			else
+				power:SetStatusBarTexture(statusbar)
+				r, g, b = t[1], t[2], t[3]
+			end
+		else
+			r, g, b = t[1], t[2], t[3]
+		end
 	end
 
 	if(b) then
```

Comment:
The variables `r`, `g` and `b` can still exist here, even if atlas is used, and will result in a discolored bar.